lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017 John Kwiatkoski
+# Copyright: (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: flatpak
+short_description: Install and remove flatpaks
+description:
+    - The flatpak module allows users to manage installation and removal of flatpaks.
+version_added: '2.6'
+author:
+- John Kwiatkoski (@jaykayy)
+requirements:
+- flatpak
+options:
+  name:
+    description:
+    - When I(state) is set to C(present), I(name) is best used as `http(s)` url format.
+      When set to C(absent) the same `http(s)` will try to remove it using the
+      name of the flatpakref. However, there is no naming standard between
+      names of flatpakrefs and what the reverse DNS name of the installed flatpak
+      will be. Given that, it is best to use the http(s) url for I(state=present)
+      and the reverse DNS I(state=absent). Alternatively reverse dns format can optimally
+      be used with I(state=absent), ex. I(name=org.gnome.gedit).
+    required: true
+  executable:
+    description:
+    - The path to the C(flatpak) executable to use.
+    default: flatpak
+  state:
+    description:
+    - Set to C(present) will install the flatpak and/or I(remote).
+    - Set to C(absent) will remove the flatpak and/or I(remote).
+    choices: [ absent, present ]
+    default: present
+  repo:
+    description:
+    - The repository to install the flatpak from
+    default: flathub
+    required: false
```

Comment:
Not strictly necessary as the default for parameter is that they're not required.